neqo-transport/src/connection/mod.rs

```diff
@@ -1896,12 +1896,12 @@ impl Connection {
         let mut tokens = Vec::new();
         let primary = path.borrow().is_primary();
 
-        let ack_token = if primary {
-            self.acks
-                .write_frame(space, now, builder, &mut self.stats.borrow_mut().frame_tx)?
-        } else {
-            None
-        };
+        if primary {
+            let stats = &mut self.stats.borrow_mut().frame_tx;
+            if let Some(t) = self.acks.write_frame(space, now, builder, stats)? {
+                tokens.push(t);
```

Comment:
Oh, with the change to use `ack_end`, we don't need to hold the ACK token back any more.  I forgot to make that change with the last patch.  This is far less brittle now.